doc/build/sysbuild/index.rst

```diff
@@ -0,0 +1,501 @@
+namespaces to direct settings either to sysbuild itself or to a specific Zephyr
+application managed by sysbuild using the information in these sections.
+
+The following example shows how to build :ref:`hello_world` with MCUboot enabled,
+applying to both images debug optimizations:
+
+.. tabs::
+
+   .. group-tab:: ``west build``
+
+      .. zephyr-app-commands::
+         :tool: west
+         :app: samples/hello_world
+         :board: reel_board
+         :goals: build
+         :west-args: --sysbuild
+         :gen-args: -DSB_CONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_DEBUG_OPTIMIZATIONS=y -Dmcuboot_DEBUG_OPTIMIZATIONS=y
+         :compact:
+
+   .. group-tab:: ``cmake``
+
+      .. zephyr-app-commands::
+         :tool: cmake
+         :app: share/sysbuild
+         :board: reel_board
+         :goals: build
+         :gen-args: -DAPP_DIR=samples/hello_world -DSB_CONFIG_BOOTLOADER_MCUBOOT=y -DCONFIG_DEBUG_OPTIMIZATIONS=y -Dmcuboot_DEBUG_OPTIMIZATIONS=y
+         :compact:
+
+See the following subsections for more information.
+
+.. _sysbuild_cmake_namespace:
+
+CMake variable namespacing
+==========================
+
+CMake variable settings can be passed to CMake using ``-D<var>=<value>`` on the
+command line. You can also set Kconfig options via CMake as
+``-DCONFIG_<var>=<value>`` or ``-D<namespace>_CONFIG_<var>=<value>``.
+
+Since sysbuild is the entry point for the build system, and sysbuild is written
+in CMake, all CMake variables are first processed by sysbuild.
+
+Sysbuild creates a namespace for each domain. The namespace prefix is the
+domain's application name. See :ref:`sysbuild_zephyr_application` for more
+information.
+
+To set the variable ``<var>`` in the namespace ``<namespace>``, use this syntax::
+
+  -D<namespace>_<var>=<value>
+
+For example, to set the CMake variable ``FOO`` in the ``my_sample`` application
+build system to the value ``BAR``, run the following commands:
+
+.. tabs::
+
+   .. group-tab:: ``west build``
+
+      ::
+
+         west build --sysbuild ... -- -Dmy_sample_FOO=BAR
+
+   .. group-tab:: ``cmake``
+
+      ::
+
+        cmake -Dmy_sample_FOO=BAR ...
+
+.. _sysbuild_kconfig_namespacing:
+
+Kconfig namespacing
+===================
+
+To set the sysbuild Kconfig option ``<var>`` to the value ``<value>``, use this syntax::
+
+  -DSB_CONFIG_<var>=<value>
+
+In the previous example, ``SB_CONFIG`` is the namespace prefix for sysbuild's Kconfig
+options.
+
+To set a Zephyr application's Kconfig option instead, use this syntax::
+
+  -D<namespace>_CONFIG_<var>=<value>
+
+In the previous example, ``<namespace>`` is the application name discussed above in
+:ref:`sysbuild_cmake_namespace`.
+
+For example, to set the Kconfig option ``FOO`` in the ``my_sample`` application
+build system to the value ``BAR``, run the following commands:
+
+.. tabs::
+
+   .. group-tab:: ``west build``
+
+      ::
+
+         west build --sysbuild ... -- -Dmy_sample_FOO=BAR
```

Comment:
Unless I misunderstood the previous explanations, I think it would be less confusing to explicit the 'CONFIG' in this example:
```suggestion
         west build --sysbuild ... -- -Dmy_sample_CONFIG_FOO=BAR
```

Reply:
Thanks for catching.
`CONFIG_` should definitely be there.